### PR TITLE
Date and date time picker

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

**Comment by Diogomartf** [comment]
It's looking really good right now.

Just a few improvements to nail it:

Do we need seconds? Seems like a bit too much.

image

We need background color when hovering on the dates so it gives feedback in which date you have your mouse on.
Something like this would work 👇   (don't show hover bg if is disabled)

image

Also on dark mode we can't tell if is disabled.

image

**Comment by luzzif** [comment]
As for the seconds I'd keep it. The other issues should be solved now.

**Comment by guerrap** [comment]
The datetime picker gets a bit crushed on mobile, maybe we could reduce the space taken by the time selection.
image

**Review by Diogomartf — APPROVED** [review]

**Review by Diogomartf — DISMISSED** [review]

File: packages/ui/src/components/input/date-time/index.tsx [hunk 1]
```diff
@@ -1,45 +1,105 @@
...
+                    !!popoverRef.current &&
+                    !popoverRef.current.contains(event.target as Node)
+                )
+                    setOpen(false);
+            };
+            document.addEventListener("mousedown", handleCloseOnClick);
+            return () => {
+                document.removeEventListener("mousedown", handleCloseOnClick);
+            };
+        }, []);
```
[the remark this answers is not in the document]
Comment (luzzif) [reply packages/ui/src/components/input/date-time/index.tsx:56]: You made me realize we're already using `react-use` here so I could just use `useClickAwayListener` from it. The same goes for the frontend actually.

**Pull request merged**